5.5/Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM centos:centos7
+
+ADD mysql /opt/openshift/
+ENV PATH /opt/openshift/bin:$PATH
+
+WORKDIR /opt/openshift
+RUN create_user.sh
+
+RUN rpmkeys --import file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7 && \
+    yum -y --setopt=tsflags=nodocs install https://www.softwarecollections.org/en/scls/rhscl/mysql55/epel-7-x86_64/download/rhscl-mysql55-epel-7-x86_64.noarch.rpm && \
+    yum -y --setopt=tsflags=nodocs install hostname mysql55 && \
+    yum clean all
+
+
+VOLUME ["/var/lib/mysql"]
+
+ENTRYPOINT ["entrypoint.sh"]
+
+EXPOSE 3306
+CMD ["mysqld_safe", "--defaults-file=./etc/my.cnf"]
```

Comment:
@soltysh it is what upstream mysql image is doing and I kinda like that. I was confused myself when I saw it for the first time. In this way, you always make sure that the entrypoint (or whatever this script will be called) handle pre-evaluation of the database creation and set the right env vars. The CMD is then explicit and you can see what is actually being run.